posts/ml-sagemaker.md

```diff
@@ -0,0 +1,201 @@
+
+// Train the model using the previously prepared test data and validate the 
+//data by providing the validation data.
+
+linear.fit({'train': s3_training_data_location})
+
+```
+
+
+Deploy the Trained Model using the Sagemaker API. Provide instance type and instance count as required. Once the deployment is complete the test data is used to test the deployed application. Once tha Model is deployed a http endpoint is generated which is used by other applications such as a lambda function as part of a streaming application or a synchronous application. To support varying  loads the application can be autoscaled.
+
+``` python
+
+linear_predictor = linear.deploy(initial_instance_count=1,
+                                 instance_type='ml.c4.xlarge',
+                                 endpoint_name='ippon-sagemaker-regression-v1')
+
+from sagemaker.predictor import csv_serializer, json_deserializer
+
+linear_predictor.content_type = 'text/csv'
+linear_predictor.serializer = csv_serializer
+linear_predictor.deserializer = json_deserializer
+
+test_dataset = pd.read_csv("Bio_Train.csv")
+test_feature_dataset = test_dataset[['Density', 'Age', 'Wt', 'Ht', 'Neck', 
+      'Chest', 'ABD', 'Hip', 'Thigh', 'Knee', 'Ankle', 'Biceps', 'Farm', 'Wrist']]
+
+test_features = np.array(test_feature_dataset.values).astype('float32')
+
+for tf in test_features:
+    result = linear_predictor.predict(tf)
+    print(result)
+
+```
+
+``` text
+Input: [  1.061  33.    211.75   73.5    40.    106.2   100.5   109.     65.8
+
+  40.6    24.     37.1    30.1    18.2  ]
+
+Prediction: {'predictions': [{'score': 20.166656494140625}]}
+```
+
+The deployed application can be invoked from a custom application using the AWS Sagemaker API's as below.
+
+``` java
+//Process input data and fetch the input for the model.
+String csvLine = "1.061,33.,211.75,73.5,40.,106.2,100.5,109.,65.8,40.6,24.,37.1,30.1,18.2";
+AmazonSageMakerRuntime client = AmazonSageMakerRuntimeClientBuilder.standard().
+				withRegion(Regions.US_EAST_1).withCredentials
+                             (new AWSStaticCredentialsProvider(creds)).build();
+		
+InvokeEndpointRequest invokeEndpointRequest = new InvokeEndpointRequest();
+ByteBuffer buf = ByteBuffer.wrap(csvLine.getBytes());
+invokeEndpointRequest.setContentType("text/csv");
+invokeEndpointRequest.setAccept("application/json");
+invokeEndpointRequest.setBody(buf);
+// provide appropriate endpoint name.
+invokeEndpointRequest.setEndpointName("ippon-sagemaker-regression-v1");		
+InvokeEndpointResult result = client.invokeEndpoint(invokeEndpointRequest);
+String responseJson = new String(result.getBody().array());
+//Add business logic based on the prediction received.
+```
+
+# Conclusion
+AWS Sagemaker provides capabilities to Author, Train, Deploy and Monitor Machine Learning applications using wide variety of algorithms, libraries and infrastructure. This can help Data Engineers and Data Scientists to come up with quality models.
```

Comment:
Good ending, try to give a call-to-action to include Ippon/have the reader do something after having learned this cool thing.